### PR TITLE
eth: fix error casting regression during database open

### PR DESCRIPTION
**Pull request by karalabe**
This PR fixes a regression introduced into the database opening where a return error was always interpreted as a `syscall.Errno`, even though it can be a lot of different things. Permission error is one where it crashes, but there could be others too.

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Wed Nov 4 13:51:25 UTC 2015_

**Comment by karalabe** [comment]
@zelig Please take a look at this PR that it still works as originally intended.

**Comment by bas-vk** [comment]
:+1:

File: eth/backend.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ const (
 var (
 	jsonlogger = logger.NewJsonLogger()
 
-	datadirInUseErrNos = []uint{11, 32, 35}
+	datadirInUseErrnos = map[uint]struct{}{11: struct{}{}, 32: struct{}{}, 35: struct{}{}}
```
Comment (fjl) [inline eth/backend.go]: This fix is nice and simple, so :+1:. However, if you have the time, please make this a `map[uint]bool` instead.
Empty structs are nice but hurt readbility in this case.

Comment (karalabe) [inline eth/backend.go]: Sure thing, will fix in a moment.

**Comment by karalabe** [comment]
@fjl @bas-vk Got even simpler with your suggestion. PTAL.

**Comment by obscuren** [comment]
:+1:

**Pull request merged**